packages/legend-query-builder/src/components/QueryBuilderResultPanel.tsx

```diff
@@ -269,75 +320,346 @@ const QueryBuilderGridResultContextMenu = observer(
+            operator,
+            filteredResult,
           ).catch(applicationStore.alertUnhandledError)
         : updateExistingPostFilterConditionNodeData(
             existingPostFilterNode,
             isFilterBy,
+            filteredResult,
+            operator,
           );
     };
 
+    const filterByOrOutValues = (isFilterBy: boolean): void => {
+      tdsState.queryBuilderState.resultState.filteredResults.forEach(
+        (filteredResult) => {
+          filterByOrOutValue(isFilterBy, filteredResult);
+        },
+      );
+    };
+
     const handleCopyCellValue = applicationStore.guardUnhandledError(() =>
       applicationStore.clipboardService.copyTextToClipboard(
-        event.current?.value,
+        event?.value?.toString() ?? '',
       ),
     );
 
     const handleCopyRowValue = applicationStore.guardUnhandledError(() =>
       applicationStore.clipboardService.copyTextToClipboard(
-        Object.values(event.current?.data).toString(),
+        tdsState.queryBuilderState.resultState
+          .findRowFromRowIndex(
+            tdsState.queryBuilderState.resultState.filteredResults[0]
+              ?.coordinates.rowIndex ?? 0,
+          )
+          .toString(),
       ),
     );
 
     return (
-      <MenuContent ref={ref}>
-        <MenuContentItem
-          onClick={(): void => {
-            filterByOrOut(true);
-          }}
-        >
-          Filter By
-        </MenuContentItem>
-        <MenuContentItem
-          onClick={(): void => {
-            filterByOrOut(false);
-          }}
-        >
-          Filter Out
-        </MenuContentItem>
-        <MenuContentDivider />
-        <MenuContentItem onClick={handleCopyCellValue}>
-          Copy Cell Value
-        </MenuContentItem>
-        <MenuContentItem onClick={handleCopyRowValue}>
-          Copy Row Value
-        </MenuContentItem>
-      </MenuContent>
+      <>
+        <MenuContent ref={ref}>
+          <MenuContentItem
+            onClick={(): void => {
+              filterByOrOutValues(true);
+            }}
+          >
+            Filter By
+          </MenuContentItem>
+          <MenuContentItem
+            onClick={(): void => {
+              filterByOrOutValues(false);
+            }}
+          >
+            Filter Out
+          </MenuContentItem>
+          <MenuContentDivider />
+          <MenuContentItem onClick={handleCopyCellValue}>
+            Copy Cell Value
+          </MenuContentItem>
+          <MenuContentItem onClick={handleCopyRowValue}>
+            Copy Row Value
+          </MenuContentItem>
+        </MenuContent>
+      </>
     );
   }),
 );
 
+const queryResultCellRenderer = observer(
+  (
+    params: DataGridCellRendererParams<{
+      resultState: QueryBuilderResultState;
+      cellExecutionResult: TDSExecutionResult;
+    }>,
+  ) => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const resultState = (params as any).resultState as QueryBuilderResultState;
```

Comment:
why these lint hacks, we did typing properly, if they are undefined, you can return `null`